docs/csharp/advanced-topics/interop/walkthrough-office-programming.md

```diff
@@ -96,6 +96,53 @@ This code demonstrates several of the features in C#: the ability to omit the `r
+
+### Why COM object cleanup is necessary
+
+COM objects in Office Interop require explicit cleanup because:
+
+- The .NET garbage collector doesn't automatically release COM objects
+- Each Excel or Word object you create holds resources that must be manually released
+- Without proper cleanup, Office applications remain running in the background
+- This applies to all COM objects: Application, Workbooks, Worksheets, Ranges, and more
+
+### Proper cleanup pattern
+
+The most reliable way to ensure COM objects are properly released is to factor out the COM object creation and usage into a separate non-inlineable method. This pattern guarantees that object references go out of scope and can be collected:
+
+:::code language="csharp" source="./snippets/OfficeWalkthrough/ThisAddIn.cs" id="ProperCleanup":::
+
+Add the following enhanced version of the `DisplayInExcel` method that includes proper COM object cleanup:
+
+:::code language="csharp" source="./snippets/OfficeWalkthrough/ThisAddIn.cs" id="DisplayWithCleanup":::
+
+This pattern ensures that:
+
+- COM objects are released even if an exception occurs
+- Object references are guaranteed to go out of scope when the core method returns
+- Excel processes don't remain orphaned in Task Manager
+- Memory is properly freed
+- The application behaves reliably in production environments
+
+For production applications, always implement this cleanup pattern for every COM object you create, including Application, Workbooks, Worksheets, Ranges, and other Office objects.
+
+### Common questions about COM object cleanup
+
+**Why can't garbage collection handle this automatically?**
+COM objects use reference counting for memory management, which is different from .NET's garbage collection. The .NET runtime creates a Runtime Callable Wrapper (RCW) around each COM object, but the RCW doesn't automatically release the underlying COM object when it's garbage collected.
+
+**Why do you use separate methods with MethodImpl(MethodImplOptions.NoInlining)?**
+The .NET JIT compiler can extend object lifetimes until the end of a method, which means local variable assignments to null aren't guaranteed to release references immediately. By factoring out COM object creation and usage into separate non-inlineable methods, you ensure that object references truly go out of scope when the method returns, allowing reliable cleanup.
+
+**Do I need to call GC.Collect() and GC.WaitForPendingFinalizers()?**
+With the separate non-inlineable method pattern, these calls are typically not necessary. The pattern ensures that COM object references go out of scope reliably when the method returns. However, you can optionally add these calls after the separate method call for additional safety in scenarios where you want to force immediate cleanup.
```

Comment:
This is not correct. 

The non-inlinable method is only needed when the user wants to deterministically cleanup all objects using  GC.Collect() and GC.WaitForPendingFinalizers().

The non-inlinable method alone does not help.

Reply:
I believe that the non-inlinable method in combination with explicit calls to GC.Collect() and GC.WaitForPendingFinalizers() is very advanced scenario that most users should not be concerned with.